### PR TITLE
add missing pacman -Syu invocation

### PR DESCRIPTION
**Pull request by itspngu**
As [outlined by fauxpark](https://github.com/qmk/qmk_firmware/pull/12921#issuecomment-841933242) in #12921, `pacman -Syu` is missing from util/install/arch.sh. This PR fixes that, see #12963.

## Description

The system-wide upgrade of packages is missing from the Arch QMK installer, as opposed to other distributions and OSes. This could lead to a lot of confusion and problems when supporting users of QMK. The Arch Linux installation script has been updated accordingly.

## Types of Changes

- [ ] Core
- [X] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* #12963

## Checklist

- [X] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [X] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by Gondolindrim** [comment]
Arch users of the world, unite!

**Comment by sigprof** [comment]
`--ignore avr-gcc` is needed to avoid upgrading avr-gcc 8.3.0 to the most recent version (ideally `IgnorePkg = avr-gcc` should be added to `pacman.conf`, but this might be too invasive to perform automatically).

**Comment by drashna** [comment]
This has merge conflicts that need to be resolved.

**Comment by itspngu** [comment]
> This has merge conflicts that need to be resolved.

Done

**Review by drashna — APPROVED** [review]

File: util/install/arch.sh [hunk 1]
```diff
@@ -9,8 +9,9 @@ _qmk_install() {
         arm-none-eabi-newlib avrdude dfu-programmer dfu-util
     sudo pacman --needed --noconfirm -U https://archive.archlinux.org/packages/a/avr-gcc/avr-gcc-8.3.0-1-x86_64.pkg.tar.xz
     sudo pacman --needed --noconfirm -S avr-libc # Must be installed after the above, or it will bring in the latest avr-gcc instead
+    sudo pacman -Syu
```
Comment (fauxpark) [inline util/install/arch.sh:12]: This should go in `_qmk_install_prepare()`.

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Pull request closed without merging**